### PR TITLE
[Assets] Assets creation transaction

### PR DESCRIPTION
**Pull request by nandofw**
- added new type of transaction: TRANSACTION_NEW_ASSET 
- implemented CNewAssetTx containing asset metadata
- added nAssetsForkBlock on chainparams
- added unit test that create a new assets

### CODE REVIEW

File: src/evo/providertx.h [hunk 1]
```diff
@@ -291,9 +291,92 @@ class CFutureTx {
...
 };
 
+class CNewAssetTx {
+public:
+    static const uint16_t CURRENT_VERSION = 1;
+
+    uint16_t nVersion{CURRENT_VERSION};// message version
+    std::string Name;
+    bool updatable = true;//if true this asset meta can be modify using assetTx update process. 
+    bool isunique = false;//true if this is asset is unique it has an identity per token (NFT flag)
```
Comment (npq7721) [inline src/evo/providertx.h]: should be isUnique

File: src/evo/providertx.h [hunk 2]
```diff
@@ -291,9 +291,92 @@ class CFutureTx {
...
+    uint8_t Decimalpoint = 0;
```
Comment (npq7721) [inline src/evo/providertx.h]: should be decimalPoint

**Review by npq7721 — CHANGES_REQUESTED** [review]
unlike future transaction we want create new asset transactions to follow the same logic as registering protx where it get store in a merkle root proof in mined block. we can create another issue to add those if it is end up taking too big for this PR

File: src/evo/providertx.h (hunk 2 above)
Comment (nandofw) [reply src/evo/providertx.h]: done

**Comment by nandofw** [comment]
- added asset metadata validation

**Review by npq7721 — APPROVED** [review]
LGTM

**Pull request merged**